### PR TITLE
[CORTX1.0] EOS-14045: CR notes and amendments

### PR DESCRIPTION
**Pull request by avrnv**
# Backend

## Problem Statement
<pre>
Story Ref (if any): EOS-14045
</pre>
## Unit testing on RPM done
<pre>
Tested manually.
</pre>
## Problem Description
<pre>
The main PR for EOS-14045 (https://github.com/Seagate/cortx-manager/pull/180) was merged before @tb-s was able to review it. This PR includes amendments required on the CR.
</pre>
## Solution
<pre>
Addressing all issues in https://github.com/Seagate/cortx-manager/pull/180
</pre>
## Unit Test Cases
<pre>
None
</pre>

### CODE REVIEW

File: csm/core/services/usl.py [hunk 1]
```diff
@@ -146,100 +145,90 @@ async def _format_lyve_pilot_access_params(self, s3_account: Dict,
...
                        s3_bucket_service: S3BucketService,
-                       s3_credentials: Optional[S3Credentials] = None,
-                       s3_account_name: Optional[str] = None,
-                       iam_user_name: Optional[str] = None,
-                       iam_user_access_key_id: Optional[str] = None,
-                       bucket_name: Optional[str] = None) -> None:
+                       s3_credentials: Optional[S3Credentials],
+                       s3_account_name: Optional[str],
+                       iam_user_name: Optional[str],
+                       iam_user_access_key_id: Optional[str],
```
Comment (tb-s) [inline csm/core/services/usl.py:161]: Is this parameter still required?

File: csm/core/services/usl.py [hunk 2]
```diff
@@ -481,12 +467,11 @@ async def post_register_device(
             await self._cleanup(
                 s3_account_service,
                 s3_iam_users_service,
-                s3_access_keys_service,
                 s3_bucket_service,
                 lyve_pilot_s3_credentials,
                 lyve_pilot_s3_account.get('account_name'),
                 lyve_pilot_iam_user.get('user_name'),
-                lyve_pilot_iam_user_access_key.get('access_key_id'),
+                lyve_pilot_iam_user.get('access_key_id'),
```
Comment (tb-s) [inline csm/core/services/usl.py:474]: We have to remove this paramater from this call as well (if it is removed from the `_cleanup` function impl).

Comment (tb-s) [reply csm/core/services/usl.py:474]: Please check line 529 of this commit, too. I can't leave a comment there for some reason.

**Review by tb-s — CHANGES_REQUESTED** [review]

File: csm/core/services/usl.py (hunk 2 above)
Comment (tb-s) [reply csm/core/services/usl.py:474]: (These comments should be one line above 🙃)

File: csm/core/services/usl.py (hunk 1 above)
Comment (avrnv) [reply csm/core/services/usl.py]: No, it is not. Removed.

File: csm/core/services/usl.py (hunk 2 above)
Comment (avrnv) [reply csm/core/services/usl.py]: Removed.

File: csm/core/services/s3/accounts.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def __init__(self, s3plugin, provisioner):
         self._s3_root_client = IamRootClient()
 
     @Log.trace_method(Log.DEBUG, exclude_args=['password'])
-    async def create_account(self, account_name: str, account_email: str, password: str):
+    async def create_account(self, account_name: str, account_email: str, password: str) -> dict:
```
Comment (tb-s) [inline csm/core/services/s3/accounts.py:48]: Almost there!

Using `dict` as a type is now supported in Python 3.9. Unfortunately it is not in Python 3.6 :/ We have to use `typing.Dict`.

Comment (avrnv) [reply csm/core/services/s3/accounts.py:48]: It's not actually my idea. I could see `dict` as a returning type almost everywhere in this file. Do you think we should fix all of them?

Comment (tb-s) [reply csm/core/services/s3/accounts.py:48]: Yes, probably. But not with this patch, let's just make this one right.

Comment (avrnv) [reply csm/core/services/s3/accounts.py]: Ok, done.

**Review by tb-s — APPROVED** [review]

**Review by kupranay — APPROVED** [review]

**Pull request merged**